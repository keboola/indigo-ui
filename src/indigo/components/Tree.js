import React from 'react';
import PropTypes from 'prop-types';

import TreeNode from './tree/TreeNode';

class Tree extends React.Component {
  render() {
    return (
      <div className="tree" onClick={this.props.onClick}>
        <TreeNode data={this.props.data} />
      </div>
    );
  }
}

Tree.propTypes = {
  data: PropTypes.object.isRequired,
  onClick: PropTypes.func
};

export default Tree;
