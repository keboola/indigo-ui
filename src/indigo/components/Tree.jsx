import React from 'react';
import PropTypes from 'prop-types';

import TreeNode from './tree/TreeNode';

export default React.createClass({
  propTypes: {
    data: PropTypes.object.isRequired,
    onClick: PropTypes.func
  },

  render() {
    return (
      <div className="kb-tree" onClick={this.props.onClick}>
        <TreeNode data={this.props.data}/>
      </div>
    );
  }

});
