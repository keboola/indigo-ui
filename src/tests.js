import renderer from 'react-test-renderer';

export const snapshot = (component) => {
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
};
