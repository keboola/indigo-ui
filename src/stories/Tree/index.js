import React from 'react';
import { storiesOf } from '@storybook/react';
import { fromJS } from 'immutable';

import Tree from '../../indigo/components/Tree';

storiesOf('Tree', module)
  .add(
    'Tree - simple structure',
    () => {
      return (
        <Tree
          data={fromJS({
            config: '337955292',
            component: 'keboola.ex-db-mysql',
            mode: 'run',
          })}
        />
      );
    },
    { info: { text: 'Render Immutable data structure as unordered list.' } }
  )
  .add(
    'Tree - advanced structure',
    () => {
      return (
        <Tree
          data={fromJS({
            config: '337955292',
            component: 'keboola.ex-db-mysql',
            mode: 'run',
            options: {
              username: 'user',
              password: 'nbusr123',
            },
            ui: {
              flags: ['deprecated', 'deleted', 'obsolete'],
            },
          })}
        />
      );
    },
    { info: { text: 'Render Immutable data structure as unordered list.' } }
  );
