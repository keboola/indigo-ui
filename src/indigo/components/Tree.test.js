import React from 'react';
import { fromJS } from 'immutable';

import { matchesSnapshot } from '../../tests';
import Tree from './Tree';

describe('<Tree />', () => {
  it('Tree - default', () => {
    matchesSnapshot(
      <Tree
        data={fromJS({
          config: '337955292',
          component: 'keboola.ex-db-mysql',
          mode: 'run'
        })}
      />
    );
  });

  it('Tree - with password', () => {
    matchesSnapshot(
      <Tree
        data={fromJS({
          config: '337955292',
          component: 'keboola.ex-db-mysql',
          mode: 'run',
          password: 'pass'
        })}
      />
    );
  });
});
