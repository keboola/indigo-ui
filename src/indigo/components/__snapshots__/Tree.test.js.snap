// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tree /> Tree - default 1`] = `
<div
  className="tree"
>
  <ul>
    <li>
      <span>
        <strong>
          config
        </strong>
        : 
        337955292
      </span>
    </li>
    <li>
      <span>
        <strong>
          component
        </strong>
        : 
        keboola.ex-db-mysql
      </span>
    </li>
    <li>
      <span>
        <strong>
          mode
        </strong>
        : 
        run
      </span>
    </li>
  </ul>
</div>
`;

exports[`<Tree /> Tree - with password 1`] = `
<div
  className="tree"
>
  <ul>
    <li>
      <span>
        <strong>
          config
        </strong>
        : 
        337955292
      </span>
    </li>
    <li>
      <span>
        <strong>
          component
        </strong>
        : 
        keboola.ex-db-mysql
      </span>
    </li>
    <li>
      <span>
        <strong>
          mode
        </strong>
        : 
        run
      </span>
    </li>
    <li>
      <span>
        <strong>
          password
        </strong>
        : 
        <span
          title="Protected content, click to reveal"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-lock fa-fw icon-clickable"
            data-icon="lock"
            data-prefix="fas"
            focusable="false"
            onClick={[Function]}
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M80 192V144C80 64.47 144.5 0 224 0C303.5 0 368 64.47 368 144V192H384C419.3 192 448 220.7 448 256V448C448 483.3 419.3 512 384 512H64C28.65 512 0 483.3 0 448V256C0 220.7 28.65 192 64 192H80zM144 192H304V144C304 99.82 268.2 64 224 64C179.8 64 144 99.82 144 144V192z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </span>
    </li>
  </ul>
</div>
`;
